packages/skygear-react-native/android/src/main/java/io/skygear/reactnative/SGSkygearReactNativeModule.java

```diff
@@ -188,6 +189,32 @@ public void signAnonymousToken(String kid, String data, Promise promise) {
         }
     }
 
+    @Override
+    public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
+        if (this.openURLPromise == null) {
+            return;
+        }
+
+        if (requestCode == REQUEST_CODE_AUTHORIZATION) {
+            if (resultCode == Activity.RESULT_CANCELED) {
+                this.openURLPromise.reject("CANCEL", "CANCEL");
```

Comment:
The detection looks good. So now for developer who want to detect cancel they will need to check `err.message === "CANCEL"`? Will we consider having SDKError class per our discussion?
Btw one minor different between android and ios, the err.code in android is `CANCAL` and ios is `EUNSPECIFIED`. We may want it to be consistent